connection/cluster/ClusterUserManager.java

```diff
@@ -73,16 +73,70 @@ public void create(String username, String password) {
         failsafeTask.runPrimaryReplica();
     }
 
+    @Override
+    public void delete(String username) {
+        ClusterClient.FailsafeTask<Void> failsafeTask = client.createFailsafeTask(
+                SYSTEM_DB,
+                parameter -> {
+                    parameter.client().stub().usersDelete(deleteReq(username));
+                    return null;
+                }
+        );
+        failsafeTask.runPrimaryReplica();
+    }
+
     @Override
     public Set<User> all() {
         ClusterClient.FailsafeTask<Set<User>> failsafeTask = client.createFailsafeTask(
                 SYSTEM_DB,
-                (parameter) ->
-                        parameter.client().stub()
-                                .usersAll(allReq()).getNamesList().stream()
-                                .map(name -> new ClusterUser(client, name))
-                                .collect(Collectors.toSet())
+                (parameter) -> {
+                    Set<User> users = new HashSet<>();
+                    List<ClusterUserProto.User> protoUsers = parameter.client().stub().usersAll(allReq()).getUsersList();
+
+                    for (ClusterUserProto.User user: protoUsers) {
+                        if (user.getPasswordExpiryOptCase() ==
+                                ClusterUserProto.User.PasswordExpiryOptCase.PASSWORDEXPIRYOPT_NOT_SET) {
```

Comment:
I recon we sohuld add a `User.of(Proto)` method to hide this `if` case away into the right domain

Reply:
Agreed, implemented.